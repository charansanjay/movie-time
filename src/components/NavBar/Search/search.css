.search {
  justify-self: center;
  border: none;
  padding: 1.1rem 1.6rem;
  font-size: 1.8rem;
  border-radius: 0.7rem;
  width: 40rem;
  transition: all 0.3s;
  color: var(--color-text);

  /* background-color: var(--color-background-900); */
  background-color: var(--color-primary-light);
}

.search::placeholder {
  color: var(--color-text-dark);
}

.search:focus {
  outline: none;
  box-shadow: 0 2.4rem 2.4rem rgba(0, 0, 0, 0.1);
  transform: translateY(-2px);
}

/* Responsive Styles */

/* For devices with width <= 1024px */
@media (max-width: 1024px) {
  .search {
    width: 30rem;
    font-size: 1.6rem;
  }
}

/* For devices with width <= 768px */
@media (max-width: 768px) {
  .search {
    width: 25rem;
    font-size: 1.6rem;
    padding: 1.2rem .8rem;
  }
}

/* For devices with width <= 480px */
@media (max-width: 480px) {
  .search {
    width: 20rem;
    font-size: 1.4rem;
    padding: .8rem .8rem;
  }
}
