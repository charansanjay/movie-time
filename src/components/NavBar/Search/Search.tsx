import { useEffect, useRef } from "react";
import "./search.css"

type SearchProps = {
  query: string;
  setQuery: (query: string) => void;
}

const Search = ({ query, setQuery }: SearchProps) => {

  const inputEl = useRef<HTMLInputElement>(null);

  /* Effects */
  // Siedeffect that focuses the input element when the component mounts
  useEffect(() => {
    if (inputEl.current) {
      inputEl.current.focus();
    }
  }, []);

  return (
    <input
      className='search'
      type='text'
      placeholder='Search movies...'
      value={query}
      onChange={(e) => setQuery(e.target.value)}
      ref={inputEl}
    />
  );
};

export default Search;
