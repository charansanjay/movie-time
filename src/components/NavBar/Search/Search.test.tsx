import {
  render,
  screen,
  fireEvent,
  cleanup,
  waitFor,
} from '@testing-library/react';
import { describe, it, expect, vi, afterEach } from 'vitest';
import '@testing-library/jest-dom';

import Search from './Search.tsx';

// Mock setQuery function
const setQuery = vi.fn();

afterEach(() => {
  vi.clearAllMocks();
  cleanup();
});

describe('Search Component', () => {
  it('should render the component without crashing', () => {
    render(<Search query='' setQuery={setQuery} />);
    expect(screen.getByPlaceholderText('Search movies...')).toBeInTheDocument();
  });

    it('should call setQuery when the input value changes', async () => {
    render(<Search query='' setQuery={setQuery} />);
    const input = screen.getByPlaceholderText('Search movies...');
    fireEvent.change(input, { target: { value: 'test' } });
    await waitFor(() => {
      expect(setQuery).toHaveBeenCalledTimes(1);
    });
  });

  it('should call setQuery with the correct value when the input value changes', async () => {
    render(<Search query='' setQuery={setQuery} />);
    const input = screen.getByPlaceholderText('Search movies...');
    fireEvent.change(input, { target: { value: 'test' } });
    await waitFor(() => {
      expect(setQuery).toHaveBeenCalledWith('test');
      expect(setQuery).toHaveBeenCalledTimes(1);
    });
  });

  it('should focus the input element when the component mounts', () => {
    render(<Search query='' setQuery={setQuery} />);
    const input = screen.getByPlaceholderText('Search movies...');
    expect(input).toHaveFocus();
  });

  /* 
  * toMatchSnapshot is a Jest matcher that compares this output to a previously saved snapshot, 
    to ensure that the Search component renders consistently over time. 
  * If the current output matches the saved snapshot, the test will pass. 
    If it does not match, the test will fail, indicating that the rendered output of the Search component has changed.
  */
  it('should match the snapshot', () => {
    const { asFragment } = render(<Search query='' setQuery={setQuery} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match the snapshot with a query value', () => {
    const { asFragment } = render(<Search query='test' setQuery={setQuery} />);
    expect(asFragment()).toMatchSnapshot();
  });

});
