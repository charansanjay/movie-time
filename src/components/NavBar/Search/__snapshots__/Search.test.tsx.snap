// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Component > should match the snapshot 1`] = `
<DocumentFragment>
  <input
    class="search"
    placeholder="Search movies..."
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`Search Component > should match the snapshot with a query value 1`] = `
<DocumentFragment>
  <input
    class="search"
    placeholder="Search movies..."
    type="text"
    value="test"
  />
</DocumentFragment>
`;
